ddl/ddl_worker.go

```diff
@@ -97,6 +97,23 @@ func (d *ddl) getFirstDDLJob(t *meta.Meta) (*model.Job, error) {
 	return job, errors.Trace(err)
 }
 
+// handleTooLargeJob handles the too large DDL job.
+func (d *ddl) handleTooLargeJob(t *meta.Meta, job *model.Job, err error) error {
+	if err == nil {
+		return nil
+	}
+	if kv.ErrEntryTooLarge.Equal(err) {
+		log.Warnf("[ddl] update DDL job %v failed %v", job, errors.ErrorStack(err))
+		// Reduce this txn entry size.
+		job.BinlogInfo.Clean()
```

Comment:
why need to clean binloginfo here?

Reply:
Because binloginfo's tableInfo is large. If doesn't clean it, we can't commit this job. @winkyao

Reply:
got it.